superset/models/core.py

```diff
@@ -590,6 +590,8 @@ def set_sqlalchemy_uri(self, uri):
         conn.password = password_mask if conn.password else None
         self.sqlalchemy_uri = str(conn)  # hides the password
 
+    @utils.memoized(
```

Comment:
The `watch` is a nice precaution (keep it!), but I think that these attrs wouldn't change during the lifecycle of the ORM instance object. Unless I'm missing something they'd live over a single web request or CLI command and get garbage collected.

Reply:
That's what I thought as well, but without `watch` a good number of unit tests were failing/in error. Those are probably changing attributes to set up the tests instead of creating new objects.